lib/internal/test_runner/test.js

```diff
@@ -183,6 +185,11 @@ class Test extends AsyncResource {
         return;
       }
 
+      if (this.shouldReportSubtest) {
+        this.reporter.subTest(this.indent, this.name);
```

Comment:
Is it possible to do this in `finalize()` instead of calling `this.reporter.subtest()` in multiple locations (we should try to keep the number of different places that log output to a minimum)? Based on [this comment](https://github.com/nodejs/node/blob/d3c579dfef1c34e8b2d348e2514dcb4fc0007a56/lib/internal/test_runner/test.js#L413-L416), I would expect it to be OK to do so.

If that works, then you can also drop the `shouldReportSubtest` field completely.

Reply:
That will result in having the subtest reported after the subtests themselves, not before them (due to line 194)